bot/on_message/bots/openai_bot.py

```diff
@@ -1,10 +1,21 @@
 import contextlib
+from dataclasses import dataclass
+from typing import Any, Optional
 import openai
 # from bot.setup.bots import WeezerpediaAPI
 
 from rich import print
 import random
 
+DEFAULT_MESSAGE_LOOKBACK_COUNT = 12
+
+@dataclass(frozen=True)
+class PromptParams:
```

Comment:
this generalizes the open ai prompt inputs beyond just the discord message class attributes